### ISSUE TITLE
Regression: v1.17 skaffold run -> Unknown flag: mount

### ISSUE THREAD

**Issue by taylortails** [GoogleContainerTools/skaffold#5207]
I use the experimental Dockerfile syntax to enable a 'mount' to cache artifacts downloaded from maven central.

skaffold works in v1.16 - but breaks in v1.17 - I guess because of new dependency calculation code.

### Expected behavior

`skaffold run` should work as v1.16

### Actual behavior

```
skaffold run...

Checking cache...
 - ...image...: Error checking cache.
failed to build: getting hash for artifact "...image...": getting dependencies for "...image...": parsing dockerfile "...path...\\Dockerfile": Dockerfile parse error line 43: Unknown flag: mount
```

### Information

- Skaffold version: v1.16 works, v1.17.0 doesn't, v1.17.2 doesn't
- Operating system: Windows

Example of Dockerfile syntax
```
# syntax=docker/dockerfile:experimental
FROM alpine

RUN --mount=id=common,sharing=private,type=cache,target=/root \
  ls
```

**Comment by briandealwis** [comment]
I think this has been fixed by #5052 as I can't reproduce at HEAD but can reproduce with 1.17.2.  Please try the bleeding-edge build at https://skaffold.dev/docs/install/

**Comment by taylortails** [comment]
you are correct, the error does not occur in bleeding-edge b5ccab5. Thanks.
